Index: connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java
===================================================================
--- connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java	(revision 61999)
+++ connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java	(working copy)
@@ -42,6 +42,7 @@
 import org.exoplatform.container.ExoContainerContext;
 import org.exoplatform.container.xml.InitParams;
 import org.exoplatform.ecm.connector.fckeditor.FCKUtils;
+import org.exoplatform.ecm.utils.text.Text;
 import org.exoplatform.portal.webui.util.SessionProviderFactory;
 import org.exoplatform.services.cms.BasePath;
 import org.exoplatform.services.cms.drives.DriveData;
@@ -193,14 +194,14 @@
       ManageableRepository manageableRepository = repositoryService.getRepository(repositoryName);
       Session session = sessionProvider.getSession(workspaceName, manageableRepository);
       ManageDriveService manageDriveService = (ManageDriveService)ExoContainerContext.getCurrentContainer()
-      	.getComponentInstanceOfType(ManageDriveService.class);
-      
-      String driverHomePath  = manageDriveService.getDriveByName(driverName, repositoryName).getHomePath();
+      	.getComponentInstanceOfType(ManageDriveService.class);      
+      String driverHomePath  = manageDriveService.getDriveByName(Text.escapeIllegalJcrChars(driverName), 
+          Text.escapeIllegalJcrChars(repositoryName)).getHomePath();
       String itemPath = driverHomePath
                         + ((currentFolder != null && !"".equals(currentFolder) && !driverHomePath.endsWith("/")) ? "/" : "")
                         + currentFolder;
       itemPath = StringUtils.replaceOnce(itemPath, "${userId}", userId);
-      Node node = (Node)session.getItem(itemPath);
+      Node node = (Node)session.getItem(Text.escapeIllegalJcrChars(itemPath));
       
       return buildXMLResponseForChildren(node, null, repositoryName, filterBy, session, currentPortal);
 
@@ -246,9 +247,10 @@
       @HeaderParam("content-type") String contentType,
       @HeaderParam("content-length") String contentLength) throws Exception {
 
-  	Node currentFolderNode = getParentFolderNode(repositoryName, workspaceName, driverName, currentFolder);
-    return createUploadFileResponse(inputStream, repositoryName, workspaceName, currentFolderNode,
-        currentPortal, jcrPath, uploadId, language, contentType, contentLength, limit);
+  	Node currentFolderNode = getParentFolderNode(repositoryName, workspaceName, Text.escapeIllegalJcrChars(driverName), 
+  	    Text.escapeIllegalJcrChars(currentFolder));
+    return createUploadFileResponse(inputStream, repositoryName, Text.escapeIllegalJcrChars(workspaceName), currentFolderNode,
+        currentPortal, Text.escapeIllegalJcrChars(jcrPath), uploadId, language, contentType, contentLength, limit);
   }
 
   /**
@@ -285,9 +287,12 @@
       @QueryParam("fileName") String fileName,
       @QueryParam("uploadId") String uploadId) throws Exception {
     try {
-    	Node currentFolderNode = getParentFolderNode(repositoryName, workspaceName, driverName, currentFolder);
-      return createProcessUploadResponse(repositoryName, workspaceName, currentFolderNode, siteName, userId, jcrPath,
-          action, language, fileName, uploadId);  
+      if ((repositoryName != null) && (workspaceName != null) && (driverName!=null) && (currentFolder!=null)) {
+        Node currentFolderNode = getParentFolderNode(repositoryName, workspaceName, Text.escapeIllegalJcrChars(driverName), 
+    	    Text.escapeIllegalJcrChars(currentFolder));
+        return createProcessUploadResponse(repositoryName, workspaceName, currentFolderNode, siteName, userId, 
+          Text.escapeIllegalJcrChars(jcrPath), action, language, fileName, uploadId);
+      }
     } catch (Exception e) {
       log.error("Error when perform processUpload: ", e);
     }
Index: web/wcmportal/src/main/webapp/fckeditor/exo/content/js/UploadForm.js
===================================================================
--- web/wcmportal/src/main/webapp/fckeditor/exo/content/js/UploadForm.js	(revision 61999)
+++ web/wcmportal/src/main/webapp/fckeditor/exo/content/js/UploadForm.js	(working copy)
@@ -39,7 +39,7 @@
 	var tblActionContainer =  eXo.core.DOMUtil.findFirstDescendantByClass(uploadForm, "table", "ActionContainer");
 	var trFolder =  eXo.core.DOMUtil.findFirstDescendantByClass(tblActionContainer, "tr", "PathFolder");
 	var spanFolder = eXo.core.DOMUtil.findDescendantsByTagName(trFolder, "span")[0];
-	spanFolder.innerHTML += ":"+ eXp.store.currentFolder;
+	spanFolder.innerHTML += ":"+ decodeURIComponent(decodeURIComponent(eXp.store.currentFolder));
 };
 
 UploadForm.prototype.showAlert = function() {
@@ -167,7 +167,7 @@
 		}
 	} else {
 		eXoWCM.UploadForm.showAlert();	
-	}
+	}	
 };
 
 UploadForm.prototype.uploadFileAbort = function() {
@@ -284,7 +284,7 @@
 	var xmlTreeNodes = eXoWCM.PluginUtils.request(connector);
 	if(!xmlTreeNodes) return;
 	var fileList = xmlTreeNodes.getElementsByTagName('File');
-	if(fileList && fileList.length > 0) eXoWCM.PluginUtils.listFiles(fileList);
+	if(fileList && fileList.length > 0) eXoWCM.PluginUtils.listFiles(fileList);	
 };
 
 UploadForm.prototype.isInvalidName = function(name) {
Index: web/wcmportal/src/main/webapp/fckeditor/exo/content/js/PluginUtils.js
===================================================================
--- web/wcmportal/src/main/webapp/fckeditor/exo/content/js/PluginUtils.js	(revision 61999)
+++ web/wcmportal/src/main/webapp/fckeditor/exo/content/js/PluginUtils.js	(working copy)
@@ -31,8 +31,8 @@
 		var id = eXoWCM.PluginUtils.generateIdDriver(nodeList[i]);
 		treeHTML += '<div class="Node" onclick="eXoWCM.PluginUtils.actionColExp(this);">';
 		treeHTML += 	'<div class="ExpandIcon">';		
-		treeHTML += 		'<a title="'+strName+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" isUpload="'+isUpload+'" onclick="eXoWCM.PluginUtils.renderBreadcrumbs(this);" name="'+strName+'" id="'+id+'">';
-		treeHTML +=			strName;	
+		treeHTML += 		'<a title="'+decodeURIComponent(strName)+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" isUpload="'+isUpload+'" onclick="eXoWCM.PluginUtils.renderBreadcrumbs(this);" name="'+decodeURIComponent(strName)+'" id="'+id+'">';
+		treeHTML +=			decodeURIComponent(strName);	
 		treeHTML +=			'</a>';
 		treeHTML += 	'</div>';			
 		treeHTML += '</div>';			
@@ -59,7 +59,7 @@
 			var driverPath = nodeList[i].getAttribute("driverPath");
 			treeHTML += '<div class="Node" onclick="eXoWCM.PluginUtils.actionColExp(this);">';
 			treeHTML += 	'<div class="ExpandIcon">';
-			treeHTML +=			'<a title="'+label+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+strName+'" id="'+id+'"  driverPath="'+driverPath+'">';
+			treeHTML +=			'<a title="'+decodeURIComponent(label)+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+decodeURIComponent(strName)+'" id="'+id+'"  driverPath="'+driverPath+'">';
 			treeHTML +=				label;	
 			treeHTML += 		'</a>';
 			treeHTML +=		'</div>';
@@ -85,8 +85,8 @@
 			var strName = nodeList[i].getAttribute("name");
 			treeHTML += '<div class="Node" onclick="eXoWCM.PluginUtils.actionColExp(this);">';
 			treeHTML += 	'<div class="ExpandIcon">';
-			treeHTML +=			'<a title="'+ strName +'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+strName+'" id="'+id+'">';
-			treeHTML +=				strName;	
+			treeHTML +=			'<a title="'+ decodeURIComponent(strName) +'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+decodeURIComponent(strName)+'" id="'+id+'">';
+			treeHTML +=				decodeURIComponent(strName);	
 			treeHTML += 		'</a>';
 			treeHTML +=		'</div>';
 			treeHTML +=	'</div>';
@@ -102,8 +102,8 @@
 				var	strName	= currentNodeList[i].getAttribute("name");
 				treeHTML += '<div class="Node" onclick="eXoWCM.PluginUtils.actionColExp(this);">';
 				treeHTML += 	'<div class="ExpandIcon">';
-				treeHTML +=			'<a title="'+strName+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+strName+'" id="'+id+'">';
-				treeHTML +=				strName;	
+				treeHTML +=			'<a title="'+decodeURIComponent(strName)+'" class="NodeIcon DefaultPageIcon" href="javascript:void(0);" onclick="getDir(this, event);" name="'+decodeURIComponent(strName)+'" id="'+id+'">';
+				treeHTML +=				decodeURIComponent(strName);	
 				treeHTML += 		'</a>';
 				treeHTML +=		'</div>';
 				treeHTML +=	'</div>';
@@ -179,7 +179,7 @@
 		if (!label) label = node;
 		var newRow = tblRWS.insertRow(i+1);
 		newRow.className = clazz;
-		newRow.insertCell(0).innerHTML = '<div class="Item '+clazzItem+'" url="'+url+'" nodeType="'+nodeType+'" onclick="eXoWCM.PluginUtils.insertContent(this);">'+label+'</div>';
+		newRow.insertCell(0).innerHTML = '<div class="Item '+clazzItem+'" url="'+url+'" nodeType="'+nodeType+'" onclick="eXoWCM.PluginUtils.insertContent(this);">'+decodeURIComponent(label)+'</div>';
 		newRow.insertCell(1).innerHTML = '<div class="Item">'+ list[i].getAttribute("dateCreated") +'</div>';
 		newRow.insertCell(2).innerHTML = '<div class="Item">'+ list[i].getAttribute("size")+'&nbsp;kb' +'</div>';
 		
@@ -320,11 +320,11 @@
 			var node = document.getElementById(currentNode.id);
 			if(node) strOnclick = "eXoWCM.PluginUtils.actionBreadcrumbs('"+node.id+"')";		
 			if(beforeNode == null) {
-				strHTML += '<a class="Nomal" href="javascript:void(0);" onclick="'+strOnclick+'">'+label+'</a>';
+				strHTML += '<a class="Nomal" href="javascript:void(0);" onclick="'+strOnclick+'">'+ decodeURIComponent(label) +'</a>';
 				tmpNode.innerHTML = strHTML;
 				breadscrumbsContainer.appendChild(tmpNode);
 			} else {
-				strHTML += '<a class="Nomal" href="javascript:void(0);" onclick="'+strOnclick+'">'+label+'</a>';
+				strHTML += '<a class="Nomal" href="javascript:void(0);" onclick="'+strOnclick+'">'+ decodeURIComponent(label) +'</a>';
 				strHTML += '<div class="RightArrowIcon"><span></span></div>';
 				tmpNode.innerHTML = strHTML;
 				breadscrumbsContainer.insertBefore(tmpNode, beforeNode);
@@ -405,16 +405,23 @@
 	var hostName = eXoPlugin.hostName;
 	var nodeType = objContent.getAttribute('nodeType');
 	var url 	= objContent.getAttribute('url');
-	var name 	= objContent.innerHTML;
+	var temp = url;
+ var index = temp.indexOf("%27");
+ while(index != -1){
+  temp = temp.replace("%27","%2527");
+		index = temp.indexOf("%27");
+ }
+ url = temp;
+	var name 	= encodeURIComponent(objContent.innerHTML);
 	var strHTML = '';	
 	if(window.opener.document.getElementById(getParameterValueByName("browserType"))){		
 		strHTML += url;		
 		window.opener.document.getElementById(getParameterValueByName("browserType")).value=strHTML;
 	} else {
 		if(nodeType.indexOf("image") >=0) {
-			strHTML += "<img src='"+url+"' name='"+name+"' alt='"+name+"'/>";
+			strHTML += "<img src=\""+url+"\" name=\""+name+"\" alt=\""+name+"\"/>";
 		} else {
-			strHTML += "<a href='" + url+"' style='text-decoration:none;'>"+name+"</a>";		
+			strHTML += "<a href=\"" + url+"\" style='text-decoration:none;'>"+name+"</a>";		
 		}
 		FCK.InsertHtml(strHTML);
 	}			
Index: web/wcmportal/src/main/webapp/fckeditor/exo/content/content.html
===================================================================
--- web/wcmportal/src/main/webapp/fckeditor/exo/content/content.html	(revision 61999)
+++ web/wcmportal/src/main/webapp/fckeditor/exo/content/content.html	(working copy)
@@ -424,7 +424,7 @@
 					currentNode.setAttribute("currentFolder", currentFolder);
 				}
 		}
-		eXp.store.currentFolder = currentFolder;
+		eXp.store.currentFolder = decodeURIComponent(currentFolder);
 		eXp.store.currentNode = currentNode;
 		driverName = eXp.store.driverName;
 		var strReplace 	= "getFoldersAndFiles?driverName="+driverName+"&currentFolder="+currentFolder+"&currentPortal="+eXoPlugin.portalName+"&" ;	
