Index: portlets/presentation/src/main/webapp/groovy/ParameterizedContentListViewer/UIPCLVContainer.gtmpl
===================================================================
--- portlets/presentation/src/main/webapp/groovy/ParameterizedContentListViewer/UIPCLVContainer.gtmpl	(revision 62314)
+++ portlets/presentation/src/main/webapp/groovy/ParameterizedContentListViewer/UIPCLVContainer.gtmpl	(working copy)
@@ -9,7 +9,7 @@
 def portletId = rcontext.getWindowId();
 portletId = isPortalAppEditting ? "EditMode-" + portletId : portletId;
 def quickEditingBlockId = portletId + "-QuickEditingBlock";
-rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
+//rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
 
 %>
 <div class="$uicomponent.id" id="$uicomponent.id">
Index: portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl
===================================================================
--- portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl	(revision 62314)
+++ portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl	(working copy)
@@ -9,12 +9,12 @@
 		def portletId = rcontext.getWindowId();
 		portletId = isPortalAppEditting ? "EditMode-" + portletId : portletId;
 		def quickEditingBlockId = portletId + "-QuickEditingBlock";
-		rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
+//		rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
 		if (Utils.isShowQuickEdit()) {					
 	%>
 			<div style="position: relative; height: 0px;" id="<%=quickEditingBlockId%>">
 				<div style="position: absolute; right: 0px; top: 0px; z-index: 1">
-					<a href="<%=uicomponent.event("QuickEdit")%>" title="<%= _ctx.appRes("UICLVContainer.title.QuickEdit");%>"><div class="QuickEdit"><span></span></div></a>					 
+				<a class="QuickEdit" href="<%=uicomponent.event("QuickEdit")%>" title="<%= _ctx.appRes("UIPCLVContainer.title.QuickEdit");%>"></a>					 
 				</div>
 			</div>
 	<% } %>		
Index: portlets/presentation/src/main/webapp/groovy/SingleContentViewer/UIPresentationContainer.gtmpl
===================================================================
--- portlets/presentation/src/main/webapp/groovy/SingleContentViewer/UIPresentationContainer.gtmpl	(revision 62314)
+++ portlets/presentation/src/main/webapp/groovy/SingleContentViewer/UIPresentationContainer.gtmpl	(working copy)
@@ -11,7 +11,7 @@
 		def portletId = rcontext.getWindowId();
 		portletId = isPortalAppEditting ? "EditMode-" + portletId : portletId;
 		def quickEditingBlockId = portletId + "-QuickEditingBlock";
-		rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
+//		rcontext.getJavascriptManager().addJavascript('eXo.wcm.insertQuickeditingBlock("' + portletId + '","' + quickEditingBlockId + '");');
 		def node = uicomponent.getNodeView();
 		if (node != null) {
 			def preferences = rcontext.getRequest().getPreferences();
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/portal/portal/acme/pages.xml
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/portal/portal/acme/pages.xml	(revision 62314)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/portal/portal/acme/pages.xml	(working copy)
@@ -8,8 +8,8 @@
     <title>Overview</title>
 	  <access-permissions>Everyone</access-permissions>
 	  <edit-permission>editor:/platform/web-contributors</edit-permission>
-	  <container id="ACMEBody" template="system:/groovy/portal/webui/container/UITableColumnContainer2.gtmpl">
-			<container id="ACMELeft" template="system:/groovy/portal/webui/container/UIContainer2.gtmpl" width="640px">
+	  <container id="ACMEBody" template="system:/groovy/wcm/webui/container/UITableColumnContainer.gtmpl">
+			<container id="ACMELeft" template="system:/groovy/portal/webui/container/UIContainer2.gtmpl">
 	  			<application>
 				    <instance-id>portal#acme:/presentation/SingleContentViewer/acme-introduce</instance-id>
 				    <title>Homepage</title>
@@ -44,7 +44,7 @@
     <title>News</title>
 	  <access-permissions>Everyone</access-permissions>
 	  <edit-permission>editor:/platform/web-contributors</edit-permission>
-	  <container id="ACMEBody" template="system:/groovy/portal/webui/container/UITableColumnContainer2.gtmpl">
+	  <container id="ACMEBody" template="system:/groovy/wcm/webui/container/UITableColumnContainer.gtmpl">
 	  	<container id="ACMECategories" template="system:/groovy/portal/webui/container/UIContainer2.gtmpl" width="200px">
 	  		<application>
 			    <instance-id>portal#acme:/presentation/CategoryNavigationPortlet/acme-categories</instance-id>
