Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl	(revision 61663)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl	(working copy)
@@ -44,7 +44,7 @@
 						<div class="CaptionItem" style="border-bottom: 0px;">
 							<% 
                                 if(imgSrc != null && uicomponent.isShowField(UICLVPortlet.SHOW_THUMBNAILS_VIEW)) {
-                                    %><img src="$imgSrc" width="270px" style="padding-bottom:10px;" /><%
+                                    %><img src="$imgSrc" alt="$imgSrc" width="270px" style="padding-bottom:10px;" /><%
                                 }
 								if (uicomponent.isShowField(UICLVPortlet.SHOW_TITLE)) {
 									%>
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl	(revision 61663)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl	(working copy)
@@ -70,11 +70,11 @@
 								if (itemDateCreated != null && uicomponent.isShowField(UICLVPortlet.SHOW_DATE_CREATED)) { 
 									%><div class="CaptionDate"><%=itemDateCreated%></div><% 
 								} 
-							%>											
+							%>																
 							<div class="CaptionContent">				
 								<% 
 									if(imgSrc != null && uicomponent.isShowField(UICLVPortlet.SHOW_THUMBNAILS_VIEW)) {
-										%><img src="$imgSrc" width="91" /><% 
+										%><img src="$imgSrc" alt="$imgSrc" width="91" /><% 
 									}
 									if (itemSummary != null && itemSummary.trim().length() != 0 && uicomponent.isShowField(UICLVPortlet.SHOW_SUMMARY)) { 
 										%><pre><%=itemSummary%></pre><% 
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl	(revision 61663)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl	(working copy)
@@ -44,7 +44,7 @@
 						<div class="CaptionItem" style="border-bottom: 0px;">
 							<% 
                                 if(imgSrc != null && uicomponent.isShowField(UICLVPortlet.SHOW_THUMBNAILS_VIEW)) {
-                                    %><img src="$imgSrc" width="30" style="float:left; padding-right:5px;" /><%
+                                    %><img src="$imgSrc" alt="$imgSrc" width="30" style="float:left; padding-right:5px;" /><%
                                 }
 								if (uicomponent.isShowField(UICLVPortlet.SHOW_TITLE)) {
 									%>
