Index: connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/PortalLinkConnector.java
===================================================================
--- connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/PortalLinkConnector.java	(revision 65279)
+++ connectors/fckeditor/src/main/java/org/exoplatform/wcm/connector/fckeditor/PortalLinkConnector.java	(working copy)
@@ -304,12 +304,15 @@
     if (page == null) {
       accessMode = PUBLIC_ACCESS;
     } else {
-        for (String role : page.getAccessPermissions()) {
+      String[] roles = page.getAccessPermissions();
+      if (roles !=null) {
+        for (String role : roles) {
           if (EVERYONE_PERMISSION.equalsIgnoreCase(role)) {
             accessMode = PUBLIC_ACCESS;
             break;
           }
-        }      
+        }
+      }      
     }
     String pageUri = "/" + servletContext.getServletContextName() + "/" + accessMode + "/" + portalName + "/" + pageNode.getUri();
 
