Index: portlets/presentation/src/main/java/org/exoplatform/wcm/webui/administration/UISiteAdministrationPortlet.java
===================================================================
--- portlets/presentation/src/main/java/org/exoplatform/wcm/webui/administration/UISiteAdministrationPortlet.java	(revision 64999)
+++ portlets/presentation/src/main/java/org/exoplatform/wcm/webui/administration/UISiteAdministrationPortlet.java	(working copy)
@@ -44,6 +44,9 @@
   /** The last locale. */
   private Locale lastLocale = null;
   
+
+  /** The Constant VISITOR. */
+  public static final int VISITOR           = -1;
   /**
    * Instantiates a new uI site administration portlet.
    * 
@@ -79,7 +82,7 @@
         adminToolbar.changeNavigationsLanguage(currentLocale.getLanguage());
         lastLocale = currentLocale;
       }
-    }        
-    super.processRender(app, context);
+      if (adminToolbar.getRole() != VISITOR) super.processRender(app, context);
+    }             
   }
 }
