Index: portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVFolderMode.java
===================================================================
--- portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVFolderMode.java	(revision 67945)
+++ portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVFolderMode.java	(working copy)
@@ -17,6 +17,7 @@
 package org.exoplatform.wcm.webui.clv;
 
 import java.util.ArrayList;
+import java.util.Calendar;
 import java.util.HashMap;
 import java.util.List;
 
@@ -80,7 +81,7 @@
     int itemsPerPage = Integer.parseInt(portletPreferences.getValue(UICLVPortlet.ITEMS_PER_PAGE, null));
     PaginatedNodeIterator paginatedNodeIterator = new PaginatedNodeIterator(nodes, itemsPerPage);
     getChildren().clear();
-    clvPresentation = addChild(UICLVPresentation.class, null, null);    
+    clvPresentation = addChild(UICLVPresentation.class, null, "UICLVPresentation-" + Calendar.getInstance().getTimeInMillis());    
     String templatePath = getFormViewTemplatePath();
     ResourceResolver resourceResolver = getTemplateResourceResolver();    
     clvPresentation.init(templatePath, resourceResolver, paginatedNodeIterator);    
Index: portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/config/UICLVConfig.java
===================================================================
--- portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/config/UICLVConfig.java	(revision 67945)
+++ portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/config/UICLVConfig.java	(working copy)
@@ -18,6 +18,7 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Calendar;
 import java.util.List;
 
 import javax.jcr.Node;
@@ -57,6 +58,7 @@
 import org.exoplatform.webui.application.portlet.PortletRequestContext;
 import org.exoplatform.webui.config.annotation.ComponentConfig;
 import org.exoplatform.webui.config.annotation.EventConfig;
+import org.exoplatform.webui.core.UIPortletApplication;
 import org.exoplatform.webui.core.lifecycle.UIFormLifecycle;
 import org.exoplatform.webui.core.model.SelectItemOption;
 import org.exoplatform.webui.event.Event;
@@ -578,7 +580,7 @@
             uiFolderViewer.init();
           } else if (newViewerMode.equals(UICLVConfig.VIEWER_MANUAL_MODE)) {                       
             uiContentListViewerPortlet.removeChild(UICLVFolderMode.class);
-            uiCorrectContentsViewer = uiContentListViewerPortlet.addChild(UICLVManualMode.class, null, null);            
+            uiCorrectContentsViewer = uiContentListViewerPortlet.addChild(UICLVManualMode.class, null, UIPortletApplication.VIEW_MODE+"-" + Calendar.getInstance().getTimeInMillis());
             uiCorrectContentsViewer.init();
             uiViewerManagementForm.doPublication(repository, workspace, portletRequestContext, selectedList, portletId);
           }
@@ -592,7 +594,7 @@
             uiViewerManagementForm.doPublication(repository, workspace, portletRequestContext, selectedList, portletId);
           } else if (newViewerMode.equals(UICLVConfig.VIEWER_AUTO_MODE)) {            
             uiContentListViewerPortlet.removeChild(UICLVManualMode.class);
-            uiFolderViewer = uiContentListViewerPortlet.addChild(UICLVFolderMode.class, null, null);
+            uiFolderViewer = uiContentListViewerPortlet.addChild(UICLVFolderMode.class, null, UIPortletApplication.VIEW_MODE+"-" +Calendar.getInstance().getTimeInMillis());
             uiFolderViewer.init();
           }
         }
Index: portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVManualMode.java
===================================================================
--- portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVManualMode.java	(revision 67945)
+++ portlets/presentation/src/main/java/org/exoplatform/wcm/webui/clv/UICLVManualMode.java	(working copy)
@@ -17,6 +17,7 @@
 package org.exoplatform.wcm.webui.clv;
 
 import java.util.ArrayList;
+import java.util.Calendar;
 import java.util.List;
 
 import javax.jcr.Node;
@@ -76,7 +77,7 @@
     ObjectPageList pageList = new ObjectPageList(nodes, itemsPerPage);    
     String templatePath = getFormViewTemplatePath();
     ResourceResolver resourceResolver = getTemplateResourceResolver();
-    UICLVPresentation clvPresentation = addChild(UICLVPresentation.class, null, null);
+    UICLVPresentation clvPresentation = addChild(UICLVPresentation.class, null, "UICLVPresentation-" + Calendar.getInstance().getTimeInMillis());
     clvPresentation.init(templatePath, resourceResolver, pageList);    
     clvPresentation.setContentColumn(portletPreferences.getValue(UICLVPortlet.HEADER, null));
     clvPresentation.setShowLink(Boolean.parseBoolean(portletPreferences.getValue(UICLVPortlet.SHOW_LINK, null)));
Index: portlets/presentation/src/main/java/org/exoplatform/wcm/webui/category/UICategoryNavigationPortlet.java
===================================================================
--- portlets/presentation/src/main/java/org/exoplatform/wcm/webui/category/UICategoryNavigationPortlet.java	(revision 67945)
+++ portlets/presentation/src/main/java/org/exoplatform/wcm/webui/category/UICategoryNavigationPortlet.java	(working copy)
@@ -64,7 +64,7 @@
     Calendar lCDateTime = Calendar.getInstance();
     addChild(UIPopupContainer.class, null, "UIPopupContainer-" + lCDateTime.getTimeInMillis());
     if (PortletMode.VIEW.equals(mode)) {
-      addChild(UICategoryNavigationTree.class, null, null);
+      addChild(UICategoryNavigationTree.class, null, "UICategoryNavigationTree-" + lCDateTime.getTimeInMillis());
     } else if (PortletMode.EDIT.equals(mode)) {
       addChild(UICategoryNavigationConfig.class, null, null);
     } 
Index: portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl
===================================================================
--- portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl	(revision 67945)
+++ portlets/presentation/src/main/webapp/groovy/ContentListViewer/UICLVContainer.gtmpl	(working copy)
@@ -2,12 +2,8 @@
 		import org.exoplatform.wcm.webui.clv.UICLVPresentation;
 		import	org.exoplatform.wcm.webui.Utils;
 		import org.exoplatform.portal.webui.util.Util; 
-		import java.util.Calendar;
-		
-		Calendar lCDateCalendar = Calendar.getInstance();
-		def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
-<div class="$uicomponent.id" id="$uicomponent.id-$timeCreated">
+<div class="$uicomponent.id" id="$uicomponent.id">
 	<%	
 		def rcontext = _ctx.getRequestContext() ;
 		def isPortalAppEditting = Util.getUIPortalApplication().isEditting();
Index: portlets/presentation/src/main/webapp/groovy/CategoryNavigation/UICategoryNavigationTree.gtmpl
===================================================================
--- portlets/presentation/src/main/webapp/groovy/CategoryNavigation/UICategoryNavigationTree.gtmpl	(revision 67945)
+++ portlets/presentation/src/main/webapp/groovy/CategoryNavigation/UICategoryNavigationTree.gtmpl	(working copy)
@@ -5,13 +5,9 @@
 	 *          chuong.phan@exoplatform.com, phan.le.thanh.chuong@gmail.com
 	 * Jun 29, 2009  
 	 */
-	 import java.util.Calendar;
-		
-	 Calendar lCDateCalendar = Calendar.getInstance();
-	 def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="$uicomponent.id">
+<div id="$uicomponent.id" class="$uicomponent.id">
 	<% 
 		uicomponent.buildTree();
 		uicomponent.renderChildren(); 
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/DocumentsTemplate.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/DocumentsTemplate.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/DocumentsTemplate.gtmpl	(working copy)
@@ -10,13 +10,9 @@
 		import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 		import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 		import org.exoplatform.wcm.webui.Utils;
-		import java.util.Calendar;
-		
-		Calendar lCDateCalendar = Calendar.getInstance();
-		def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="DocumentsCLVTemplate">
+<div id="$uicomponent.id" class="DocumentsCLVTemplate">
 	<div class="CLV">
 		<%
 			def header = uicomponent.getHeader();
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/OneColumnCLVTemplate.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/OneColumnCLVTemplate.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/OneColumnCLVTemplate.gtmpl	(working copy)
@@ -10,13 +10,9 @@
 		import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 		import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 		import org.exoplatform.wcm.webui.Utils;
-		import java.util.Calendar;
-		
-		Calendar lCDateCalendar = Calendar.getInstance();
-		def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="OneColumnCLVTemplate">
+<div id="$uicomponent.id" class="OneColumnCLVTemplate">
 	<div class="CLV">
 		<%
 			def header = uicomponent.getHeader();
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationBigImage.gtmpl	(working copy)
@@ -3,13 +3,9 @@
 	import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 	import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 	import org.exoplatform.wcm.webui.Utils;
-	import java.util.Calendar;
-		
-	Calendar lCDateCalendar = Calendar.getInstance();
-	def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="UICLVPresentation">
+<div id="$uicomponent.id" class="UICLVPresentation">
 	<%	
 		String header = uicomponent.getHeader();
 		if (uicomponent.isShowHeader() && header != null && header.trim().length() != 0 ) {
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationDefault.gtmpl	(working copy)
@@ -3,12 +3,8 @@
 	import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 	import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 	import org.exoplatform.wcm.webui.Utils;
-	import java.util.Calendar;
-		
-	Calendar lCDateCalendar = Calendar.getInstance();
-	def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
-<div id="$uicomponent.id-$timeCreated" class="UICLVPresentation">
+<div id="$uicomponent.id" class="UICLVPresentation">
 	<%	
 		String header = uicomponent.getHeader();
 		if (uicomponent.isShowHeader() && header != null && header.trim().length() != 0 ) {
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/UIContentListPresentationSmall.gtmpl	(working copy)
@@ -3,13 +3,9 @@
 	import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 	import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 	import org.exoplatform.wcm.webui.Utils;
-	import java.util.Calendar;
-		
-	Calendar lCDateCalendar = Calendar.getInstance();
-	def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="UICLVPresentation">
+<div id="$uicomponent.id" class="UICLVPresentation">
 	<%	
 		String header = uicomponent.getHeader();
 		if (uicomponent.isShowHeader() && header != null && header.trim().length() != 0 ) {
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/TwoColumnsCLVTemplate.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/TwoColumnsCLVTemplate.gtmpl	(revision 67945)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/application-templates/content-list-viewer/list-by-folder/TwoColumnsCLVTemplate.gtmpl	(working copy)
@@ -10,13 +10,9 @@
 		import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
 		import org.exoplatform.wcm.webui.clv.UICLVPortlet;
 		import org.exoplatform.wcm.webui.Utils;
-		import java.util.Calendar;
-		
-		Calendar lCDateCalendar = Calendar.getInstance();
-		def timeCreated = lCDateCalendar.getTimeInMillis();
 %>
 
-<div id="$uicomponent.id-$timeCreated" class="TwoColumnsCLVTemplate">
+<div id="$uicomponent.id" class="TwoColumnsCLVTemplate">
 	<div class="CLV">
 		<%
 			def header = uicomponent.getHeader();
