Index: webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/content/one/UIContentBrowsePanelOne.java
===================================================================
--- webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/content/one/UIContentBrowsePanelOne.java	(revision 68720)
+++ webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/content/one/UIContentBrowsePanelOne.java	(working copy)
@@ -106,7 +106,6 @@
       wcmPublicationService.enrollNodeInLifecycle(realNode, portalOwner, remoteUser);          
     }
     
-    wcmPublicationService.updateLifecyleOnChangeContent(realNode, portalOwner, remoteUser, null);
     if (!Utils.isEditPortletInCreatePageWizard()) {
       String pageId = Util.getUIPortal().getSelectedNode().getPageReference();
       UserPortalConfigService upcService = getApplicationComponent(UserPortalConfigService.class);
