Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/CategoryView.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/CategoryView.gtmpl	(revision 49714)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/CategoryView.gtmpl	(working copy)
@@ -10,6 +10,7 @@
    	 */
 %>
 <%
+    import java.net.URLEncoder;
 	import java.util.Calendar;
 	import java.util.Date;
   	import javax.jcr.Node;
@@ -291,7 +292,7 @@
 	          		if (!uicomponent.isSystemWorkspace()) rightClickMenu = contextMenu.getJSOnclickShowPopup(preferenceWS + ":" + Utils.formatNodeName(nodePath), uicomponent.getActionsList(data)) ;	          
 	      
 		      		%>
-			      		<div <%=rightClickMenu%> class="RowView Normal" locked="$isLocked" workspacename="$preferenceWS" objectId="$nodePath" ondblclick="$actionLink">
+			      		<div <%=rightClickMenu%> class="RowView Normal" locked="$isLocked" workspacename="$preferenceWS" objectId="<%=URLEncoder.encode(Utils.formatNodeName(nodePath), "utf-8")%>" ondblclick="$actionLink">
 				      		$strActs
 			        		<div class="LeftColumn" style="padding: 2px 0px; 	border-right: none;">
 								<div title="$name" class="Icon16x16 default16x16Icon <%=Utils.getNodeTypeIcon(data, "16x16Icon")%>">
Index: web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/ContentView.gtmpl
===================================================================
--- web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/ContentView.gtmpl	(revision 49714)
+++ web/wcmportal/src/main/webapp/WEB-INF/conf/wcm/artifacts/views/templates/ecm-explorer/ContentView.gtmpl	(working copy)
@@ -11,6 +11,7 @@
    	 */
 %>
 <%
+    import java.net.URLEncoder;
 	import java.util.Calendar;
 	import java.util.Date;
   	import javax.jcr.Node;
@@ -294,7 +295,7 @@
 	          		if (data.isLocked()) isLocked = "true";    
 	          		if (!uicomponent.isSystemWorkspace()) rightClickMenu = contextMenu.getJSOnclickShowPopup(preferenceWS + ":" + Utils.formatNodeName(nodePath), uicomponent.getActionsList(data)) ;	          
 		      		%>
-	  		<div <%=rightClickMenu%> class="RowView Normal" locked="$isLocked" workspacename="$preferenceWS" objectId="$nodePath" ondblclick="$actionLink">
+	  		    <div <%=rightClickMenu%> class="RowView Normal" locked="$isLocked" workspacename="$preferenceWS" objectId="<%=URLEncoder.encode(Utils.formatNodeName(nodePath), "utf-8")%>" ondblclick="$actionLink">
 	      		$strActs
 	    		<div class="LeftColumn" style="padding: 2px 0px; 	border-right: none;">
 					<div title="$name" class="Icon16x16 default16x16Icon <%=Utils.getNodeTypeIcon(data, "16x16Icon")%>">
