Index: webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/UISelectPathPanel.java
===================================================================
--- webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/UISelectPathPanel.java	(revision 71433)
+++ webui/wcm/src/main/java/org/exoplatform/wcm/webui/selector/UISelectPathPanel.java	(working copy)
@@ -28,12 +28,10 @@
 import javax.jcr.nodetype.NodeType;
 
 import org.exoplatform.commons.utils.ObjectPageList;
+import org.exoplatform.ecm.utils.text.Text;
 import org.exoplatform.ecm.webui.tree.selectone.UIOneNodePathSelector;
 import org.exoplatform.ecm.webui.utils.Utils;
 import org.exoplatform.portal.webui.container.UIContainer;
-import org.exoplatform.portal.webui.util.Util;
-import org.exoplatform.portal.webui.workspace.UIMaskWorkspace;
-import org.exoplatform.ecm.utils.text.Text;
 import org.exoplatform.services.cms.BasePath;
 import org.exoplatform.services.ecm.publication.PublicationService;
 import org.exoplatform.services.jcr.RepositoryService;
@@ -46,7 +44,6 @@
 import org.exoplatform.services.wcm.publication.WCMPublicationService;
 import org.exoplatform.wcm.webui.selector.content.UIContentBrowsePanel;
 import org.exoplatform.web.application.ApplicationMessage;
-import org.exoplatform.webui.application.WebuiRequestContext;
 import org.exoplatform.webui.config.annotation.ComponentConfig;
 import org.exoplatform.webui.config.annotation.EventConfig;
 import org.exoplatform.webui.core.UIApplication;
@@ -439,8 +436,7 @@
       UISelectPathPanel uiSelectPathPanel = event.getSource();      
       UIContainer uiTreeSelector = uiSelectPathPanel.getParent();
       String value = event.getRequestContext().getRequestParameter(OBJECTID);
-	  String[] values = value.split("/");
-      value = value.replaceAll(values[values.length - 1], Text.escapeIllegalJcrChars(values[values.length - 1]));
+	  value = Text.escapeIllegalJcrChars(value);
       if(uiTreeSelector instanceof UIOneNodePathSelector) {
         if(!((UIOneNodePathSelector)uiTreeSelector).isDisable()) {
           value = ((UIOneNodePathSelector)uiTreeSelector).getWorkspaceName() + ":" + value ;
Index: webui/wcm/src/main/resources/groovy/wcm/webui/selector/content/UIContentTreeBuilder.gtmpl
===================================================================
--- webui/wcm/src/main/resources/groovy/wcm/webui/selector/content/UIContentTreeBuilder.gtmpl	(revision 71433)
+++ webui/wcm/src/main/resources/groovy/wcm/webui/selector/content/UIContentTreeBuilder.gtmpl	(working copy)
@@ -4,6 +4,7 @@
   import org.exoplatform.wcm.webui.selector.content.UIContentTreeNode ;
   import org.exoplatform.web.application.JavascriptManager;
   import java.util.Date;
+  import org.exoplatform.ecm.utils.text.Text;
   
   def rcontext = _ctx.getRequestContext() ;
 	JavascriptManager jsmanager = rcontext.getJavascriptManager();
@@ -70,7 +71,7 @@
 																																	
 								objChild.innerHTML += '<div class="NoneSelectNodeIcon" onclick="' + restAction + '" id="'+
 																			eXo.wcm.MyUIWCMTreeView.constParentNode + firstId+'" treePath="'+treePath+'">'+
-																			'<div class="DefaultPageIcon">'+name+'</div>'+
+																			'<div class="DefaultPageIcon">'+decodeURI(name)+'</div>'+
 																			'</div>'+
 																			'<div class="ViewChildNodes" id="'+eXo.wcm.MyUIWCMTreeView.constChildNode + firstId+'"></div>';
 							}
@@ -126,7 +127,7 @@
 			newDiv.id = eXo.wcm.MyUIWCMTreeView.constBreadcumbsElement + i;
 			if(i <= 1) newDiv.className ="FirstNode";
 			else newDiv.className ="RightArrowIcon";
-			if(i != 0) newDiv.innerHTML = nodeNames[i];
+			if(i != 0) newDiv.innerHTML = decodeURI(nodeNames[i]);
 			else newDiv.innerHTML = "";
 			if(objParent){
 				if(objParent.getAttribute("onclick") && objParent.getAttribute("onclick") != null && objParent.getAttribute("onclick") != "null")
@@ -201,7 +202,7 @@
 		else restAction = "eXo.wcm.MyUIWCMTreeView.selectNode('"+id+"')";
 		%>
 		<div onclick="$restAction;$action" treePath="<%=treeNode.getTreePath()%>" style="" id="UIWCMTreeParentNode$id" class="NoneSelectNodeIcon">
-			<div class="DefaultPageIcon"><%= treeNode.getName() %></div>
+			<div class="DefaultPageIcon"><%=Text.unescapeIllegalJcrChars(treeNode.getName()) %></div>
 		</div>
 		<div id="UIWCMTreeChildNode$id" class="ViewChildNodes">
 	<%
