Index: portlets/searches/src/main/java/org/exoplatform/wcm/webui/search/UISearchResult.java
===================================================================
--- portlets/searches/src/main/java/org/exoplatform/wcm/webui/search/UISearchResult.java	(revision 64999)
+++ portlets/searches/src/main/java/org/exoplatform/wcm/webui/search/UISearchResult.java	(working copy)
@@ -325,7 +325,15 @@
 		String workspace = portletPreferences.getValue(UIWCMSearchPortlet.WORKSPACE, null);
 		String baseURI = portletRequest.getScheme() + "://"	+ portletRequest.getServerName() + ":" + String.format("%s", portletRequest.getServerPort());
 		String basePath = portletPreferences.getValue(UIWCMSearchPortlet.BASE_PATH, null);
-		link = baseURI + portalURI + basePath + "/" + repository + "/" + workspace + node.getPath();
+		link = baseURI + portalURI + basePath + "/" + repository + "/" + workspace;
+		if (node.isNodeType("nt:frozenNode")){
+			String uuid = node.getProperty("jcr:frozenUuid").getString();
+			Node originalNode = node.getSession().getNodeByUUID(uuid);
+			link += originalNode.getPath()+ "?version=" + node.getParent().getName();
+			
+		}else {
+			link += node.getPath();
+		}
 		return link;
 	}
 
